src/Stripe.net/Infrastructure/StringOrObject.cs

```diff
@@ -4,20 +4,21 @@ namespace Stripe.Infrastructure
     using Newtonsoft.Json.Linq;
 
     internal static class StringOrObject<T>
-        where T : StripeEntityWithId
+        where T : IStripeEntityWithId
     {
         public static void Map(object value, Action<string> updateId, Action<T> updateObject)
         {
             if (value is JObject)
             {
-                T item = ((JToken)value).ToObject<T>();
+                // T item = ((JToken)value).ToObject<T>();
+                T item = Mapper<T>.MapFromJson(value.ToString());
                 updateId(item.Id);
                 updateObject(item);
             }
             else if (value is string)
             {
                 updateId((string)value);
-                updateObject(null);
+                updateObject(default(T));
```

Comment:
can you clarify what this line does/changes?

Reply:
In practice, nothing. `default(T)` returns the default value for the type `T`. For reference types, the default value is `null`.

However, since `T` is now constrained to be an interface rather than a class, the compiler complains that it cannot convert from `null` to `T`. This is (I think) because `struct`s are value types but can implement interfaces.